src/lib.rs

```diff
@@ -90,13 +91,18 @@ pub struct ZWaveManager {
 }
 
 impl ZWaveManager {
-    fn new(manager: manager::Manager) -> ZWaveManager {
-        ZWaveManager {
+    fn new(manager: manager::Manager) -> (Self, mpsc::Receiver<ZWaveNotification>) {
+        let (tx, rx) = mpsc::channel();
+
+        let manager = ZWaveManager {
             watcher: ZWaveWatcher {
                 state: Arc::new(Mutex::new(State::new())),
+                sender: Arc::new(Mutex::new(tx))
```

Comment:
I think this deserves a comment since it isn't intially clear why the Arx<Mutex<>> is needed